ETL/main.py

```diff
@@ -0,0 +1,196 @@
+    columns = data.columns
+    new_columns = []
+    for column in columns:
+        new_columns.append(str(column).replace(
+            "fields.", "").replace(" ", "_").replace("-", ""))
+    data.columns = new_columns
+    print(new_columns)
+    # Drop and Create the table
+    sql = "DROP TABLE IF EXISTS {}; ".format(table_name)
+    sql += "CREATE TABLE {} (".format(table_name)
+    for name in data.columns:
+        sql += "{} TEXT".format(name)
+        if name != list(data.columns)[-1]:
+            sql += ','
+    sql += "); "
+    cursor.execute(sql)
+    db.commit()
+
+    # Iterate through each row and add values
+    sql = "INSERT INTO {} (".format(table_name) + \
+        ','.join(data.columns) + ") VALUES "
+    allrowvalues = []
+    for index, row in data.iterrows():
+        currentrowvalues = []
+        for col in data.columns:
+            # Get the values for the current row and sanitize them into clean string data
+            currentrowvalues.append(
+                "'" + str(row[col]).replace("'", "").replace("[", "").replace("]", "").replace("nan", "") + "'")
+        allrowvalues.append(','.join(currentrowvalues))
+    for item in allrowvalues:
+        sql += "({})".format(item)
+        if item != list(allrowvalues)[-1]:
+            sql += ','
+    cursor.execute(sql)
+    db.commit()
+
+
+log("Python ETL Script Start")
+
+# --------- PHASE 1: Import from Airtables --------- #
+
+log("Import the listings airtable")
+listings_airtable = Airtable(AIRTABLE_BASE_ID, 'listings', AIRTABLE_API_KEY)
+listings = listings_airtable.get_all()
+create_import_table('etl_import_1', listings)
+del listings_airtable
+del listings
+log("Import the phone airtable")
+phone_airtable = Airtable(AIRTABLE_BASE_ID, 'phone', AIRTABLE_API_KEY)
+phone = phone_airtable.get_all()
+create_import_table('etl_import_2', phone)
+del phone_airtable
+del phone
+log("Import the address airtable")
+address_airtable = Airtable(AIRTABLE_BASE_ID, 'address', AIRTABLE_API_KEY)
+address = address_airtable.get_all()
+create_import_table('etl_import_3', address)
+del address_airtable
+del address
+log("Import the contacts airtable")
+contacts_airtable = Airtable(
+    AIRTABLE_BASE_ID, 'contacts', AIRTABLE_API_KEY)
+contacts = contacts_airtable.get_all()
+create_import_table('etl_import_4', contacts)
+del contacts_airtable
+del contacts
+log("Import the contacts airtable")
+parent_airtable = Airtable(
+    AIRTABLE_BASE_ID, 'parent_organization', AIRTABLE_API_KEY)
+parent = parent_airtable.get_all()
+create_import_table('etl_import_5', parent)
+del parent_airtable
+del parent
+
+# --------- PHASE 2: Merge multiple tables into a single staging table --------- #
+
+log("Merge the import tables")
+query = ("select etl_merge_import_tables();")
+cursor.execute(query)
+db.commit()
+
+# --------- PHASE 3: Geocode addresses --------- #
+
+log("Geocode addresses")
+geolocator = Nominatim(user_agent="rose-city-resource")
```

Comment:
Confused as to how this user agent works.

Reply:
It passes the value as the User-Agent when the library makes HTTP requests. The usage policy for the library requires setting it to the name of the application.

Reply:
Added a comment to the Python script

Reply:
Awesome thank you!